src/Parsers/ASTColumnsTransformers.cpp

```diff
@@ -95,19 +96,34 @@ void ASTColumnsExceptTransformer::formatImpl(const FormatSettings & settings, Fo
         settings.ostr << ")";
 }
 
-void ASTColumnsExceptTransformer::transform(ASTs & nodes) const
+void ASTColumnsExceptTransformer::transform(ASTs & nodes, std::vector<String> databases) const
 {
     std::set<String> expected_columns;
+
     if (original_pattern.empty())
     {
         for (const auto & child : children)
             expected_columns.insert(child->as<const ASTIdentifier &>().name());
 
+        bool have_database_name = !databases.empty();
+        std::vector<String>::iterator db = databases.begin();
+
         for (auto it = nodes.begin(); it != nodes.end();)
         {
             if (const auto * id = it->get()->as<ASTIdentifier>())
             {
-                auto expected_column = expected_columns.find(id->shortName());
+                String full_name = id->name();
+                // match table.column pattern
+                auto expected_column = expected_columns.find(full_name);
+
+                // match column pattern
+                if (expected_column == expected_columns.end())
+                    expected_column = expected_columns.find(id->shortName());
+
+                // match db.table.column pattern
+                if (expected_column == expected_columns.end() && have_database_name)
+                    expected_column = expected_columns.find(*db + "." + full_name);
```

Comment:
Simple string concatenation can be incorrect, because database name and table name can contain dot character.